source/external/cJSON.c

```diff
@@ -177,844 +183,848 @@ static void * CJSON_CDECL internal_realloc(void *pointer, size_t size)
 {
-  double maxVal = fabs(a) > fabs(b) ? fabs(a) : fabs(b);
-  return (fabs(a - b) <= maxVal * DBL_EPSILON);
+    double maxVal = fabs(a) > fabs(b) ? fabs(a) : fabs(b);
+    return (fabs(a - b) <= maxVal * DBL_EPSILON);
 }
 
 /* Render the number nicely from the given item into a string. */
 static cJSON_bool print_number(const cJSON * const item, printbuffer * const output_buffer) // NOLINT
 {
-  unsigned char *output_pointer = NULL;
-  double d = item->valuedouble;
-  int length = 0;
-  size_t i = 0;
-  unsigned char number_buffer[26] = {0}; /* temporary buffer to print the number into */
-  unsigned char decimal_point = get_decimal_point(); // NOLINT
-  double test = 0.0;
-
-  if (output_buffer == NULL)
-  {
-      return false;
-  }
-
-  /* This checks for NaN and Infinity */
-  if (isnan(d) || isinf(d))
-  {
-      length = snprintf((char*)number_buffer, sizeof(number_buffer) / sizeof(char), "null");
-  }
-  else
-  {
-      /* Try 15 decimal places of precision to avoid nonsignificant nonzero digits */
-      length = snprintf((char*)number_buffer, sizeof(number_buffer) / sizeof(char), "%1.15g", d);
-
-      /* Check whether the original double can be recovered */
-      if ((sscanf((char*)number_buffer, "%lg", &test) != 1) || !compare_double((double)test, d))
-      {
-          /* If not, print with 17 decimal places of precision */
-          length = snprintf((char*)number_buffer, sizeof(number_buffer) / sizeof(char), "%1.17g", d);
-      }
-  }
-
-  /* sprintf failed or buffer overrun occurred */
-  if ((length < 0) || (length > (int)(sizeof(number_buffer) - 1)))
-  {
-      return false;
-  }
-
-  /* reserve appropriate space in the output */
-  output_pointer = ensure(output_buffer, (size_t)length + sizeof(""));
-  if (output_pointer == NULL)
-  {
-      return false;
-  }
-
-  /* copy the printed number to the output and replace locale
-   * dependent decimal point with '.' */
-  for (i = 0; i < ((size_t)length); i++)
-  {
-      if (number_buffer[i] == decimal_point)
-      {
-          output_pointer[i] = '.';
-          continue;
-      }
-
-      output_pointer[i] = number_buffer[i];
-  }
-  output_pointer[i] = '\0';
-
-  output_buffer->offset += (size_t)length;
-
-  return true;
+    unsigned char *output_pointer = NULL;
+    double d = item->valuedouble;
+    int length = 0;
+    size_t i = 0;
+    unsigned char number_buffer[26] = {0}; /* temporary buffer to print the number into */
+    unsigned char decimal_point = get_decimal_point(); // NOLINT
+    double test = 0.0;
+
+    if (output_buffer == NULL)
+    {
+        return false;
+    }
+
+    /* This checks for NaN and Infinity */
+    if (isnan(d) || isinf(d))
+    {
+        length = snprintf((char*)number_buffer, sizeof(number_buffer) / sizeof(char), "null");
+    }
+	else if (d == (double)item->valueint) // NOLINT
+	{
+        length = snprintf((char*)number_buffer, sizeof(number_buffer) / sizeof(char), "%d", item->valueint);
+	}
```

Comment:
bad lint? Not sure why we haven't removed NOLINT in this case

Reply:
aha, it's white space vs tab.